qemu/tests/uefishell.py

```diff
@@ -0,0 +1,156 @@
+import os
+import re
+import logging
+
+from avocado.utils import process
+
+from virttest import storage
+from virttest import data_dir
+from virttest import utils_misc
+from virttest import env_process
+
+
+class UEFIShellTest(object):
+    """
+    Provide basic functions for uefishell test. To use UefiShell.iso
+    which is provided by ovmf package. Boot your VM with this as
+    the CD-ROM image and it should boot into the UEFI shell.
+    """
+
+    def __init__(self, test, params, env):
+        """
+        Init the default values for UEFIShell object
+        :param test: Kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+        self.test = test
+        self.params = params
+        self.env = env
+        self.session = None
+
+    def setup(self, under_fs0):
+        """
+        Pre-process for uefishell environment, launch vm from the
+        UefiShell.iso, and setup a serial session for uefishell
+
+        :param under_fs0: most uefi command executed under fs0:\
+        """
+        params = self.params
```

Comment:
Why not interact `self.params` directly? What is the reason assign it to `param`?